src/sdk/strategy/cbd-strategy.ts

```diff
@@ -84,18 +87,18 @@ export class CbdStrategy {
     const conditionSetEquals =
       this.conditionSet && other.conditionSet
         ? this.conditionSet.equals(other.conditionSet)
-        : false;
+        : this.conditionSet === other.conditionSet;
     return this.cohort.equals(other.cohort) && conditionSetEquals;
   }
 }
 
 export class DeployedCbdStrategy {
   constructor(
-    public cohort: Cohort,
-    public dkgRitual: DkgRitual,
-    public encrypter: Enrico,
-    public decrypter: CbdTDecDecrypter,
-    public conditionSet?: ConditionSet
+    public readonly cohort: Cohort,
+    public readonly dkgRitual: DkgRitual,
+    public readonly encrypter: Enrico,
+    public readonly decrypter: CbdTDecDecrypter,
```

Comment:
Removal of these constructor parameters should be addressed in #222 i.e. they aren't needed.